### ISSUE TITLE
Unexpected error with 3.2.2

### ISSUE THREAD

**Issue by Bovive** [farmOS/farmOS#848]
**Describe the bug**
Updating from 3.2.1 to 3.2.2 (Docker)e results in an error when accessing the site:
"The website encountered an unexpected error. Try again later."

**To Reproduce**
1. Update to 3.2.2
2. Go to site.
3. See error

**Expected behavior**
Access site without error.

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser Edge and Firefox

**Additional context**
Rolling back to 3.2.1 resolves the issue. Updating back to 3.2.2 results in the error again.
Error from the log:

`[php:notice] [pid 19] [client 172.17.0.1:51980] Symfony\\Component\\DependencyInjection\\Exception\\ServiceCircularReferenceException: Circular reference detected for service "authentication_subscriber", path: "authentication_subscriber -> authentication -> authentication_collector -> simple_oauth.authentication.simple_oauth". in /opt/drupal/web/core/lib/Drupal/Component/DependencyInjection/Container.php on line 147 #0 /opt/drupal/web/core/lib/Drupal/Component/EventDispatcher/ContainerAwareEventDispatcher.php(105): Drupal\\Component\\DependencyInjection\\Container->get('authentication_...')\n#1 /opt/drupal/vendor/symfony/http-kernel/HttpKernel.php(239): Drupal\\Component\\EventDispatcher\\ContainerAwareEventDispatcher->dispatch(Object(Symfony\\Component\\HttpKernel\\Event\\ExceptionEvent), 'kernel.exceptio...')\n#2 /opt/drupal/vendor/symfony/http-kernel/HttpKernel.php(91): Symfony\\Component\\HttpKernel\\HttpKernel->handleThrowable(Object(ArgumentCountError), Object(Symfony\\Component\\HttpFoundation\\Request), 1)\n#3 /opt/drupal/web/modules/simple_oauth/src/HttpMiddleware/BasicAuthSwap.php(54): Symfony\\Component\\HttpKernel\\HttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#4 /opt/drupal/web/core/lib/Drupal/Core/StackMiddleware/Session.php(58): Drupal\\simple_oauth\\HttpMiddleware\\BasicAuthSwap->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#5 /opt/drupal/web/core/lib/Drupal/Core/StackMiddleware/KernelPreHandle.php(48): Drupal\\Core\\StackMiddleware\\Session->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#6 /opt/drupal/web/core/lib/Drupal/Core/StackMiddleware/ContentLength.php(28): Drupal\\Core\\StackMiddleware\\KernelPreHandle->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#7 /opt/drupal/web/core/lib/Drupal/Core/StackMiddleware/ReverseProxyMiddleware.php(48): Drupal\\Core\\StackMiddleware\\ContentLength->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#8 /opt/drupal/web/core/lib/Drupal/Core/StackMiddleware/NegotiationMiddleware.php(51): Drupal\\Core\\StackMiddleware\\ReverseProxyMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#9 /opt/drupal/web/core/lib/Drupal/Core/StackMiddleware/AjaxPageState.php(36): Drupal\\Core\\StackMiddleware\\NegotiationMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#10 /opt/drupal/web/core/lib/Drupal/Core/StackMiddleware/StackedHttpKernel.php(51): Drupal\\Core\\StackMiddleware\\AjaxPageState->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#11 /opt/drupal/web/core/lib/Drupal/Core/DrupalKernel.php(704): Drupal\\Core\\StackMiddleware\\StackedHttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#12 /opt/drupal/web/index.php(19): Drupal\\Core\\DrupalKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request))\n#13 {main}

**Comment by paul121** [comment]
A similar issue was opened for the drupal/simple_oauth project: https://www.drupal.org/project/simple_oauth/issues/3438218

Have you run update.php after updating your install? Do you have any other Drupal contrib modules enabled?

**Comment by mstenta** [comment]
@Bovive Weird. I've experienced this issue myself, but only in my local development environment, and before 3.2.2 was released. I have not experienced it with 3.2.2 during the upgrades I've performed (but I will let you know if I do with any future ones).

In the context of my local development environment, it happens once in a while, and always seems to resolve itself when I restart my Docker containers. I still haven't figured out the root cause, but my suspicion was that it is due to the way I'm mounting volumes and using Composer to symlink local repositories. Are you doing that by any chance?

> Rolling back to 3.2.1 resolves the issue. Updating back to 3.2.2 results in the error again.

The fact that you're able to reproduce this makes me very curious though... Have you tried a fresh installation? Does it happen then too? Or only when you are updating an existing?

I'd also like to know the answer to @paul121's questions:

> Have you run update.php after updating your install? Do you have any other Drupal contrib modules enabled?
